### ISSUE TITLE
Add workflow for running benches in CI

### ISSUE THREAD

**Issue by alexfertel** [OpenZeppelin/rust-contracts-stylus#169]
Create a new workflow similar to the `e2e-tests` workflow that executes `cargo run -p benches`.

~~Blocked by #168~~ No longer blocked!

Ideally, this workflow also takes care of displaying the result as a comment in PRs, but this should be a separate issue.

**Comment by kaustubh1106** [comment]
hey @alexfertel I would like to work on this issue

**Comment by alexfertel** [comment]
Sure! Go for it
